src/braket/default_simulator/linalg_utils.py

```diff
@@ -35,17 +36,40 @@
 
 _QUBIT_THRESHOLD = nb.int32(10)
 
-DIAGONAL_GATES = {
-    "pauli_z",
-    "rz",
-    "s",
-    "si",
-    "phaseshift",
-    "cphaseshift",
-    "cphaseshift01",
-    "cphaseshift00",
-    "cphaseshift10",
-}
+DIAGONAL_GATES = frozenset(
+    {
+        "pauli_z",
+        "s",
+        "si",
+        "t",
+        "ti",
+        "rz",
+        "phaseshift",
+        "cz",
+        "cphaseshift",
+        "cphaseshift01",
+        "cphaseshift00",
+        "cphaseshift10",
+        "zz",
+    }
+)
+
+TWO_QUBIT_GATE_DISPATCH = MappingProxyType(
```

Comment:
Nice, I'm going to start using these. In fact, we probably no longer need to use `ImmutableDict` (not even sure we still do)